pkg/nrtupdater/nrtupdater.go

```diff
@@ -61,11 +64,12 @@ func (mi MonitorInfo) UpdateReason() string {
 	return RTEUpdateReactive
 }
 
-func NewNRTUpdater(args Args, tmconf TMConfig) *NRTUpdater {
+func NewNRTUpdater(k8sInterface kubernetes.Interface, args Args, tmconf TMConfig) *NRTUpdater {
```

Comment:
I think is better if the `NRTUpdater` takes and uses a more limited interface like
```
type NodeGetter interface {
	Get(ctx context.Context, name string, opts metav1.GetOptions) (*v1.Node, error)
}
```
This is not just to fulfill the principle of minium privilege, but also because I'm reluctant to require this functionality a Get to a node for each update. I'm even wary about adding a watch.
So I'd start decoupling the responsabilities. The NRTUpdater just needs a way to access the node metadata (uid...) from name. Another component should be responsible to pull this data and keep it uptodate (or not)

Reply:
Makes sense. 
I have tried to implement it in a newly added commit